src/utils.ts

```diff
@@ -10,14 +10,56 @@ import {execSync} from 'child_process';
 import * as mkdirp from 'mkdirp';
 const isWsl = require('is-wsl');
 
-export function defaults<T>(val: T | undefined, def: T): T {
+export enum LaunchErrorCodes {
+  ERR_LAUNCHER_PATH_NOT_SET = 'ERR_LAUNCHER_PATH_NOT_SET',
+  ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY = 'ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY',
+  ERR_LAUNCHER_UNSUPPORTED_PLATFORM = 'ERR_LAUNCHER_UNSUPPORTED_PLATFORM',
+  ERR_LAUNCHER_NOT_INSTALLED = 'ERR_LAUNCHER_NOT_INSTALLED',
+}
+
+export function defaults<T>(val: T|undefined, def: T): T {
   return typeof val === 'undefined' ? def : val;
 }
 
 export async function delay(time: number) {
   return new Promise(resolve => setTimeout(resolve, time));
 }
 
+export class LauncherError extends Error {
+  message: string;
+  code: string;
+  constructor(message: string = 'Unexpected error', code?: string) {
+    super();
+    this.message = message;
+    if (code) {
+      this.code = code;
+    }
+    this.stack = new Error().stack;
+    return this;
+  }
+}
+
+export class ChromePathNotSetError extends LauncherError {
+  message: string =
+      'The environment variable CHROME_PATH must be set to executable of a build of Chromium version 54.0 or later.';
+  code: string = LaunchErrorCodes.ERR_LAUNCHER_PATH_NOT_SET;
+}
+
+export class InvalidUserDataDirectoryError extends LauncherError {
+  message: string = 'userDataDir must be false or a path.';
+  code: string = LaunchErrorCodes.ERR_LAUNCHER_INVALID_USER_DATA_DIRECTORY;
+}
+
+export class UnsupportedPlatformError extends LauncherError {
+  message: string = `Platform ${getPlatform()} is not supported.`;
+  code: string = LaunchErrorCodes.ERR_LAUNCHER_UNSUPPORTED_PLATFORM;
+}
+
+export class ChromeNotInstalledError extends LauncherError {
+  message: string = 'No Chrome installations found.';
```

Comment:
: string can be inferred on all of these.

Reply:
@samccone Fixed in fc0d4b9.